### ISSUE TITLE
Remove locale based parsing/representation from uom-se

### ISSUE THREAD

**Issue by JeanPhiPhi** [unitsofmeasurement/uom-se#102]
Advanced string representations and parsing are useful. But they are not core UOM business. Please provide them separately.

Advanced string related helpers might be provided in a separate "uom-string" lib.

**Comment by keilw** [comment]
Will run it by @otaviojava but from my feeling we'll obey to any JDK/Java SE specific practice here. java.util.Date, Calendar or the new java.time package come with extensive formatting/Locale-aware parsing, and it's all part of "jdk-base". From all we know this will also be the same in Java 9 with a Jigsaw module.

**Comment by JeanPhiPhi** [comment]
New JavaFX Node class is nearly

**Comment by keilw** [comment]
Nearly what?

**Comment by JeanPhiPhi** [comment]
Seriously not all of what is done in Java is a good idea. For instance new JavaFX Node class is nearly one thousand lines of code. Don't think it is an example to follow :P

**Comment by keilw** [comment]
You have to differentiate between "value agnostic" elements including the likes of Collection or base types like String, etc. and value-oriented types like Money, Quantity or Date/Time.

**Comment by keilw** [comment]
Neither JSR 354 nor 363 repeat some errors or plenty of redundancies you'd find in JavaFX or java.time as well as other (older and newer) Date/Time libraries in the JDK, but reusing some of its underlying data standards looks reasonable.

**Comment by JeanPhiPhi** [comment]
Ok, so, is there any advantage to provide to/form string representations inside uom-se rather than in a separate lib?

**Comment by keilw** [comment]
It's a question of reasonable complexity. Special UnitFormat implementations like UCUM already got moved to a separate module, but doing so for just LocalFormat looks like an overkill. Again, given modularity and optionality in the API we may take advantage of that and let you chose from modules but Java 9 first has to offer full modularity. Until then I don't see a benefit in banning Locale/ResourceBundle support from the SE port. What's in Java SE (and makes sense in the context of Units) should be supported. You are also not forced to use Lambdas in SE 8, nor are you forced to use LocalUnitFormat if SimpleUnitFormat is sufficient for your needs.

**Comment by JeanPhiPhi** [comment]
TBH I don't see any complexity, just some minor code refactor. There is always a benefit to separate business from its representations. Note that JDK also deals with serialization, Corba, files, sockets and so on. Do uom-se? ;)

You're the boss!

As a final word, for our immediate need, as long as uom-se does not require any external lib and runs fine in an OSGi environment that's fine!

**Comment by keilw** [comment]
And again, EBNFUnitFormat also uses Java's Locale/ResourceBundle, so does UCUM, but they do so with "pseudo-locales" like "Print" or "CI" which in UCUM's case are simply the variants in the spec. It is not for l10n. Whether or not a UnitFormat implementation supports l10n depends on the isLocaleSensitive() flag. Should the CLDR evolve and provide far more units than the Units class does over time, there could be room for an extended "cldr-system" or similar most likely in https://github.com/unitsofmeasurement/uom-systems, not uom-lib, but those units in uom-se should rather get full CLDR support (for the languages/countries Java also does in other cases) in uom-se.

So e.g. for languages CLDR (and JDK) supports, either LocalFormat or a possible follow-up should format existing units like Units.HOUR along the lines of http://cldr.unicode.org/translation/plurals, but unless we see demand, we may not add every new unit the CLDR adds to uom-se. That would/should be up to 3rd party libraries. Take the si-units extentions ;-) Will leave it up to @otaviojava what he thinks about it, but the idea of uom-se is not a 1:1 equivalent of the RI just offering BigDecimal (which is also using Java SE where needed, so you could get more precise numeric results if you need them at the cost of more memory or processing time)

**Comment by keilw** [comment]
Since Java 9 aims to be "somewhat compatible" with OSGi, but its module system won't work exactly the same way, I think it'll be premature splitting uom-se into modules right now. Maybe if we come closer to 1.0 we'll revisit that. And see https://github.com/unitsofmeasurement/uom-se/blob/master/pom.xml other than javax.annotation, there is no other runtime dependency in uom-se other than Java SE 8.

**Comment by keilw** [comment]
As a coincidence, Java SE 9 was just delayed by another 6 months yesterday;-)

**Comment by keilw** [comment]
Discussing this with the EG, I found this information by the JDK i18n group: http://openjdk.java.net/groups/i18n/. As of Java 6 10 languages were fully supported.
According to localedata.jar Java SE 8 now for for some or all UI elements already supports 43 languages. Depending on available resources (both data and contributors) we may have to find a compromise between the 10 and 43, but I already did a full translation of Swing to 5 additional languages in 1999 http://sdj.sourceforge.net/java/locale/index.html when Sun only supported English for at least another 3-4 years, so I am confident we can support some, at least those 10 Java 6 did. Some translations already exist for 6 languages other than English as of now. So leaving modularization or separation of e.g. "localedata" into their own JARs aside, I'll close this ticket.

**Comment by keilw** [comment]
Since another forum-post drew my attention to a label/symbol not in line with many standards for Units.KILOMETRE_PER_HOUR, I created https://github.com/unitsofmeasurement/uom-se/issues/103 and the related JIRA ticket for the RI. It touches https://java.net/jira/browse/UNITSOFMEASUREMENT-161 which I reopened for discussion. Regardless of a locale or language (though in JSR 275 the JavaDoc once suggested other use cases) a method like alias() would allow to properly parse e.g. "ms" as well as "millis" if that's what your app should do. The only way the locale gets involved is via a dedicated UnitFormat implementation like LocalUnitFormat instanciated for a particular locale.
